tests/anderson_test.py

```diff
@@ -0,0 +1,198 @@
+    tol = 1e-4
+    aa = AndersonAcceleration(rot, history_length=10, maxiter=400, ridge=1e-7, tol=tol)
+    sol, state = aa.run(x0, theta, c)
+    self.assertLess(state.error, tol)
+    self.assertArraysAllClose(sol, c.astype(dtype=jnp.float32), atol=1e-3)
+
+  def test_gradient_descent(self):
+    """ Fixed point of map T(x)=x-grad_f(x) corresponding to gradient step.
+        f is chosen to be some least square regression on dummy dataset.
+    """
+    X, y = datasets.make_regression(n_samples=10, n_features=3, random_state=0)
+    least_square_grad = jax.grad(objective.least_squares)
+    def T_map(param, eta, *args, **kwargs):
+      g = least_square_grad(param, *args, **kwargs)
+      step = tree_scalar_mul(eta, g)
+      return tree_sub(param, step)
+    w_init = jnp.zeros(X.shape[1])
+    eta = 1e-1  # reasonable step size
+    tol = 1e-5
+    aa = AndersonAcceleration(T_map, history_length=5, maxiter=300, ridge=1e-5, tol=tol)
+    sol, state = aa.run(w_init, eta, (X, y))
+    self.assertLess(state.error, tol)
+    grad = least_square_grad(sol, (X,y))
+    zero = jnp.zeros_like(w_init)
+    self.assertArraysAllClose(grad, zero, atol=1e-4)
+
+  def test_filter_leaves(self):
+    """ Test support of `filter_leaves` functionnality.
+    """
+    def f(params, r):
+      x, y = params
+      fx = jnp.cos(x)  # fixed point 0.739085
+      fy = r * y * (1 - y)  # logistic map: chaotic behavior
+      return fx, fy
+    r = jnp.array([3.95])  # for this value there is chaotic behavior
+    x0logistic = jnp.array([1.]), jnp.array([0.6])
+    tol = 1e-5
+    aa = AndersonAcceleration(f, history_length=15, maxiter=300, ridge=1e-6, tol=tol, filter_leaves=(True,False))
+    sol, state = aa.init(x0logistic, r)
+    sols = []
+    for i in range(10):
+      sol, state = aa.update(sol, state, r)
+      sols.append(sol)
+    self.assertLess(state.error, tol)
+    self.assertArraysAllClose(sol[0], jnp.array([0.739085]))
+
+  def test_picard_ode(self):
+    """ Solve ODE with Picard method.
+    """
+    def f(ti, phi):
+      return 1 + phi ** 2
+    def integrate(phi_cur, ti, y0, dx):
+      f_phi = f(ti, phi_cur)
+      f_phi = jnp.concatenate([jnp.zeros((1,)), f_phi[:-1]])
+      phi_next = jnp.cumsum(f_phi) * dx + y0
+      return phi_next
+    y0 = 0
+    num_interpolating_points = 500
+    t0 = jnp.array(0.)
+    tmax = 0.6 * (jnp.pi / 2)
+    dx = (tmax - t0) / (num_interpolating_points-1)
+    phi0 = jnp.zeros(num_interpolating_points)
+    ti = np.linspace(t0, tmax, num_interpolating_points)
+    sols = [phi0]
+    aa = AndersonAcceleration(integrate, history_length=5, maxiter=300, ridge=1e-6)
+    sol, state = aa.init(phi0, ti, y0, dx)
+    sols.append(sol)
+    for k in range(aa.maxiter):
+      sol, state = aa.update(phi0, state, ti, y0, dx)
+      sols.append(sol)
+    best_approx = sols[-1]
+    tan_t = jnp.tan(ti)
+    self.assertArraysAllClose(best_approx, tan_t, atol=1e-2)
+
+  def test_simple_vjp(self):
```

Comment:
Your code call neither `jvp` nor `vjp` so the function name and the docstring are strange.